OpenRA.Mods.Common/Traits/Power/Player/PowerManager.cs

```diff
@@ -140,11 +141,17 @@ public void TriggerPowerOutage(int totalTicks)
 
 		void UpdatePowerOutageActors()
 		{
-			var actors = self.World.ActorsHavingTrait<AffectedByPowerOutage>()
-				.Where(a => !a.IsDead && a.IsInWorld && a.Owner == self.Owner);
+			var traitPairs = self.World.ActorsWithTrait<AffectedByPowerOutage>()
+				.Where(p => !p.Actor.IsDead && p.Actor.IsInWorld && p.Actor.Owner == self.Owner);
 
-			foreach (var a in actors)
-				UpdateActor(a);
+			foreach (var p in traitPairs)
+				p.Trait.UpdateStatus(p.Actor);
+		}
+
+		void IResolveOrder.ResolveOrder(Actor self, Order order)
+		{
+			if (order.OrderString == "PowerOutage")
```

Comment:
Please check for `developerMode.Enabled` here to prevent any shenanigans with the order stream.